cosmosdb/sql/ps-container-large-partition-key.ps1

```diff
@@ -0,0 +1,62 @@
+# Create a Cosmos SQL API container with large partition key support (version 2)
+$apiVersion = "2015-04-08"
+$location = "West US 2"
+$resourceGroupName = "myResourceGroup"
+$accountName = "mycosmosaccount"
+$databaseName = "database1"
+$containerName = "container1"
+$databaseResourceName = $accountName + "/sql/" + $databaseName
+$containerResourceName = $accountName + "/sql/" + $databaseName + "/" + $containerName
+$accountResourceType = "Microsoft.DocumentDb/databaseAccounts"
+$databaseResourceType = "Microsoft.DocumentDb/databaseAccounts/apis/databases"
+$containerResourceType = "Microsoft.DocumentDb/databaseAccounts/apis/databases/containers"
+
+$locations = @(
+    @{ "locationName"="West US 2"; "failoverPriority"=0 },
+    @{ "locationName"="East US 2"; "failoverPriority"=1 }
+)
+
+$CosmosDBProperties = @{
+    "databaseAccountOfferType"="Standard";
+    "locations"=$locations
+}
+
+New-AzResource -ResourceType $accountResourceType `
```

Comment:
Fails with the following error:

```
New-AzResource : {"code":"BadRequest","message":"DatabaseAccount name 'mycosmosaccount-westus2' already exists.\r\nActivityId: 
67a3eafc-572b-40e3-bec5-06909fec46ce, Microsoft.Azure.Documents.Common/2.4.0.0"}
At C:\Users\sttramer\src\azure-docs-powershell-samples\cosmosdb\sql\ps-container-large-partition-key.ps1:24 char:1  
+ New-AzResource -ResourceType $accountResourceType `
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzResource], ErrorResponseMessageException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceCmdlet
 ```

No such DB exists in my account anywhere so I assume this is required to be a unique identifier. Please fix.

This applies to all DB creation.

Reply:
How am I supposed to fix that? You used a cosmos account name that is already being used.

Replace the account name with something more unique than appending the region name and rerun.

Reply:
You have checked the following box in the __Required__ list:

>  All identifiers which must be universally unique are guaranteed to be so.

There is no indication that the account name is a user-supplied value, since it has what may be considered a "reasonable default". You are required to make this value as unique as possible (collisions are always possible, after all.) The recommendation is to use a random identifier as part of any unique name. For PowerShell, this is the `Get-Random` cmdlet.